src/Components/Common/DateInputV2.tsx

```diff
@@ -310,7 +315,7 @@ const DateInputV2: React.FC<Props> = ({
                             className="aspect-square w-[14.26%]"
                           >
                             <div
-                              onClick={setDateValue(d)}
+                              onClick={setDateValue(d, close)}
```

Comment:
You could've done something like this instead of passing the close function and requiring to add a type for it right?

```suggestion
                              onClick={() => {
                                setDateValue(d);
                                close();
                              }}
```

Reply:
@rithviknishad Initially I solved it the way you suggested. The problem with this approach is, the original code calls `setDateValue` function while passing to onClick: `onClick={setDateValue(d)}` which runs while the component is rendered. 

By doing this we get the green highlight on the previously mentioned date which is done with the help of 
` isSelectedDate(d)` function.
![Screenshot 2023-05-26 134841](https://github.com/coronasafe/care_fe/assets/72242181/aaf78837-b1e6-4392-9bc6-a8aa7cc115d3)

By using the way you suggested the green box will gray out initially when user will click the input.

![Screenshot 2023-05-26 135252](https://github.com/coronasafe/care_fe/assets/72242181/15765760-8721-4bc9-b774-23776390d76b)

Reply:
Alright, thanks @sachdevavaibhav. Nice work!